lib/galaxy/webapps/galaxy/api/repositories.py

```diff
@@ -0,0 +1,260 @@
+import json
```

Comment:
There is a lot of overlap with `lib/galaxy/webapps/galaxy/api/tool_shed_repositories.py`, was this supposed to be a rename ? All the new methods here seem to be available in that file as well.

Reply:
Good point. tool_shed_repositories.py and toolshed.py have several overlapping functions but right now tools are installed by making at least 2 individual calls see: https://github.com/galaxyproject/galaxy/blob/dev/client/galaxy/scripts/mvc/toolshed/repository-view.js#L303. prepareInstall and doInstall. One requesting from an api endpoint and the latter from the admin toolshed controller. As of right now the new repositories api is an attempt to collect and refactor required features while reducing the dependency on toolshed modules. Overall the goal is to deprecate/consolidate tool_shed_repositories.py and tool_shed.py endpoints.

Reply:
That isn't necessary for installations though; I don't know why the UI does what it does.
You can certainly install repositories with a single POST, that's how bioblend and by extension ephemeris install repos. I'm not saying the current API is particularly easy to use, but it's widely used and functional.

Happy to see where you're taking it with this, once it's ready we can probably merge the methods.

Reply:
Yes that sounds good!

Reply:
I removed the repositories api, using existing endpoints for now regardless of their dependency on toolshed modules.

Reply:
Thank you!